### PR TITLE
My big reorg

### PR DESCRIPTION
**Pull request by spaetz**
Oh no :-). I have just rewritten Epos to not pass json around and also to be able to delete a specific product from any order (e.g. the first or the third item "Hamburger") rather than just searching for its product name.
Also, it assigns "Product" objects to each "Order" and not a clear-text string which encodes some json.

This is a change all around, I have not made many nice commits out of it. But perhaps you like what you see and we can clean up the stuff and merge parts of it, bit by bit.

I did not fix migration files, as I said, I simply dropped those without regard to backward compatability. So to test this you need to recreate a database from scratch.

### CODE REVIEW

**Comment by Epse** [comment]
Will have a look at it. One of the major reasons I store serialized Products in the Orders is for archival.
Orders are supposed to stay in the database for archival, but what if the price of a Product changes?
To avoid having to keep an archive of old prices, I just give each Order a copy of the Product.

Don't know if you took that into account, will have a look around in the code for sure

**Comment by Epse** [comment]
Major props for the multiple open orders idea though

File: pos/models.py [hunk 1]
```diff
@@ -35,18 +42,60 @@ def to_json(self):
...
+        ordering = ['-lastChange']
+        get_latest_by = "lastChange"
+
+
+class ProductInOrder(models.Model):
+    """Intermediary model connecting products and orders"""
+    product = models.ForeignKey(Product, on_delete=models.CASCADE)
+    order = models.ForeignKey(Order, on_delete=models.CASCADE)
 
+
```
Comment (Epse) [inline pos/models.py]: There is whitespace on this line.

File: pos/views.py [hunk 2]
```diff
@@ -40,107 +41,108 @@ def login(request):
...
-    return HttpResponse(template.render(context, request))
+    return render(request, 'pos/addition.html', context=context)
+
+
+@login_required
+def remove_from_order(request, order_id=None, pio_id=None):
+    """Remove a specific item from the order"""
+
+    # attempt ?pio_id= paramter if it was not in the url
+    if pio_id is None: pio_id = request.GET.get('pio_id', None)
```
Comment (Epse) [inline pos/views.py]: Please just put a newline after the colon

File: pos/views.py [hunk 3]
```diff
@@ -40,107 +41,108 @@ def login(request):
...
     }
-    return HttpResponse(template.render(context, request))
+    return render(request, 'pos/addition.html', context=context)
+
+
+@login_required
+def remove_from_order(request, order_id=None, pio_id=None):
+    """Remove a specific item from the order"""
+
+    # attempt ?pio_id= paramter if it was not in the url
```
Comment (Epse) [inline pos/views.py]: Comments should have a capital if it is a sentence

File: pos/views.py [hunk 4]
```diff
@@ -40,107 +41,108 @@ def login(request):
...
+    if pio_id is None: pio_id = request.GET.get('pio_id', None)
+
+    pio = get_object_or_404(ProductInOrder, id=pio_id)
+    #sanity check order id if it matches, sth was weird with the url otherwise
```
Comment (Epse) [inline pos/views.py]: Comments should follow `# Sentence.` Capital and space are important for readability. There are other comments that have issues, have a look at the codeclimate report

**Comment by spaetz** [comment]
I see, changing prices will change prices on old orders too, that is correct. Mmmh.
-- 
Sent from mobile device. Please excuse brevity and tpyos.

**Comment by Epse** [comment]
I'm closing this one for lack of progress (and partial merge). The requested functionality is on it's way

**Pull request closed without merging**